tools/spatyper/spatyper.xml

```diff
@@ -0,0 +1,62 @@
+<tool id="spatyper" name="Generate spa type identification" version="@VERSION@+galaxy0" profile="21.05">
+    <macros>
+        <token name="@VERSION@">0.3.3</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@VERSION@">spatyper</requirement>
+        <requirement type="package" version="3.4.3">python</requirement>
+    </requirements>
+    <command> spaTyper -f $input1 -r $input2 > $output1 </command>
```

Comment:
# 1. Command element

The command-line used to execute the tool should be wrapped with `<![CDATA[ ]]>` and the `<command>` element should include the `exit_code` attribute. For example:

```xml
<command detect_errors="aggressive"><![CDATA[
    spaTyper -f $input1 -r $input2 > $output1
]]></command>
```

Using `CDATA` allows you to directly use certain characters without escaping in XML, as described here https://planemo.readthedocs.io/en/latest/standards/docs/best_practices/tool_xml.html#command-formatting

Using `detect_errors` defines how an error in the tool is detected. The `"aggressive"` option means it reports an error if either the exit code is non-zero or if the STDERR contains strings like `Error:` or `Exception:`. If this causes Galaxy to report errors when there is non, then you can change this to `detect_errors="exit_code"` to only use the exit code value. This is described here https://docs.galaxyproject.org/en/latest/dev/schema.html#error-detection

# 2. Input/output/text variables

Additionally, please surround input/output/text variables with single quotes and curly braces `'{ }'`. For example:

```
spaTyper -f '${input1}' -r '${input2}' > '${output1}'
```

This way, if there are spaces or other special characters, the executed command will still work. For example:

```
spaTyper -f 'input file with spaces.fasta' -r 'input 2 with spaces.fasta'  > 'output with spaces.fasta'
```